### PR TITLE
Logging

### LINKED ISSUE #788: Logging functionality for jpsreport
Remove old logging functions and use spdlog` like `jpscore`

### PR DESCRIPTION
**Pull request by chraibi**
Remove old logging style from jpsreport.

The geometry in jpsreport still depends on the IO/Outputhandler.

Closes #788 and #688 (item 2)

### CODE REVIEW

**Comment by JetteSchumann** [comment]
PR is still in progress. Old Logger objects are still part of the code.
We must think about the possibility to write logging information to a file as well. This is important for complex analysis and documentation purposes. Is this possible with the spdlogger?

**Comment by Ozaq** [comment]
Oh and sorry for merging already :/

File: jpsreport/methods/Method_D.cpp [hunk 1]
```diff
@@ -380,7 +380,7 @@ void Method_D::GetProfiles(
 
     FILE * Prf_velocity;
     if((Prf_velocity = Analysis::CreateFile(Prfvelocity)) == nullptr) {
-        Log->Write("cannot open the file <%s> to write the field data", Prfvelocity.c_str());
+        LOG_ERROR("cannot open the file <%s> to write the field data", Prfvelocity.c_str());
```
Comment (chraibi) [inline jpsreport/methods/Method_D.cpp]:
```suggestion
        LOG_ERROR("cannot open the file {} to write the field data", Prfvelocity.c_str());
```

File: jpsreport/geometry/Building.cpp [hunk 2]
```diff
@@ -169,11 +169,10 @@ Room * Building::GetRoom(int index) const
 {
     //todo: obsolete since the check is done by .at()
     if(_rooms.count(index) == 0) {
-        Log->Write(
-            "ERROR: Wrong 'index' in CBuiling::GetRoom() Room ID: %d size: %d",
-            index,
+        LOG_ERROR(
+            "Wrong 'index' in CBuiling::GetRoom() Room ID: {} size: {}", index,
```
Comment (chraibi) [inline jpsreport/geometry/Building.cpp:173]: clang-format  wrong

File: jpsreport/methods/Method_B.cpp [hunk 3]
```diff
@@ -48,7 +48,7 @@ Method_B::~Method_B() {}
 
 bool Method_B::Process(const PedData & peddata)
 {
```
Comment (chraibi) [inline jpsreport/methods/Method_B.cpp:50]: clang-format

File: jpsreport/methods/Method_D.cpp (hunk 1 above)
Comment (JetteSchumann) [reply jpsreport/methods/Method_D.cpp]: was already fixed

File: jpsreport/geometry/Building.cpp (hunk 2 above)
Comment (JetteSchumann) [reply jpsreport/geometry/Building.cpp]: done. forgot to reformat...

File: jpsreport/methods/Method_B.cpp (hunk 3 above)
Comment (JetteSchumann) [reply jpsreport/methods/Method_B.cpp:50]: done

**Comment by JetteSchumann** [comment]
looks like I have an issue with clangformat and Clion. working on it

File: jpsreport/Analysis.h [hunk 4]
```diff
@@ -32,21 +32,20 @@
...
 #define ANALYSIS_H_
 
-#include "IO/OutputHandler.h"
 #include "general/ArgumentParser.h"
 #include "general/Macros.h"
 #include "geometry/Building.h"
 #include "methods/ConfigData_DIJ.h"
 #include "methods/PedData.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/Analysis.h:41]: Why include Logger.h here and not in Analysis.cpp?

File: jpsreport/CMakeLists.txt [hunk 5]
```diff
@@ -39,9 +38,8 @@ set (header_files
     methods/Method_I.h
     methods/Method_J.h
     methods/ConfigData_DIJ.h
-    IO/OutputHandler.h
-    general/ArgumentParser.h
-    general/Macros.h
+        general/ArgumentParser.h
```
Comment (zeroset) [inline jpsreport/CMakeLists.txt:41]: Please indent according to the lines above.

File: jpsreport/general/ArgumentParser.h [hunk 6]
```diff
@@ -33,6 +33,7 @@
 #include "../tinyxml/tinyxml.h"
 #include "Macros.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/general/ArgumentParser.h:36]: Please include in cpp file.

File: jpsreport/geometry/Building.h [hunk 7]
```diff
@@ -36,6 +36,7 @@
 #include "Room.h"
 #include "Transition.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/geometry/Building.h:39]: Please include in cpp file.

File: jpsreport/geometry/Crossing.h [hunk 8]
```diff
@@ -32,6 +32,9 @@
 #include "Hline.h"
 #include "NavLine.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/geometry/Crossing.h:35]: Please include in cpp file.

File: jpsreport/geometry/Goal.h [hunk 9]
```diff
@@ -31,6 +31,7 @@
 
 #include "Point.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/geometry/Goal.h:34]: Please include in cpp file. ;)

File: jpsreport/geometry/Hline.h [hunk 10]
```diff
@@ -32,6 +32,8 @@
 #include "NavLine.h"
 #include "Room.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/geometry/Hline.h:35]: Please include in cpp file.

File: jpsreport/geometry/Line.h [hunk 11]
```diff
@@ -29,18 +29,11 @@
 #ifndef _LINE_H
 #define _LINE_H
 
-#include "../IO/OutputHandler.h"
 #include "Point.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/geometry/Line.h:34]: Please include in cpp file.

File: jpsreport/geometry/Obstacle.h [hunk 12]
```diff
@@ -29,6 +29,7 @@
 #ifndef OBSTACLE_H_
 #define OBSTACLE_H_
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/geometry/Obstacle.h:32]: Please include in cpp file.

File: jpsreport/geometry/Room.h [hunk 13]
```diff
@@ -30,6 +30,7 @@
 
 #include "../general/Macros.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/geometry/Room.h:33]: Please include in cpp file

File: jpsreport/geometry/SubRoom.h [hunk 14]
```diff
@@ -32,6 +32,7 @@
 
 #include "Wall.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/geometry/SubRoom.h:35]: Please include in cpp file.

File: jpsreport/geometry/Transition.h [hunk 15]
```diff
@@ -31,6 +31,7 @@
 
 #include "Crossing.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/geometry/Transition.h:34]: Please include in cpp file.

File: jpsreport/geometry/Wall.h [hunk 16]
```diff
@@ -31,6 +31,8 @@
 
 #include "Line.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/geometry/Wall.h:34]: Please include in cpp file.

File: jpsreport/methods/Method_A.h [hunk 17]
```diff
@@ -28,12 +28,12 @@
 #ifndef METHOD_A_H_
 #define METHOD_A_H_
 
-#include "../IO/OutputHandler.h"
 #include "../general/Macros.h"
 #include "../tinyxml/tinyxml.h"
 #include "MeasurementArea.h"
 #include "PedData.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/methods/Method_A.h:36]: Please include in cpp file.

File: jpsreport/methods/Method_B.h [hunk 18]
```diff
@@ -32,6 +32,7 @@
 #include "Method_C.h"
 #include "PedData.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/methods/Method_B.h:35]: Please include in cpp file.

File: jpsreport/methods/Method_C.h [hunk 19]
```diff
@@ -31,6 +31,8 @@
 #include "../Analysis.h"
 #include "PedData.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/methods/Method_C.h:34]: Please include in cpp file.

File: jpsreport/methods/Method_D.h [hunk 20]
```diff
@@ -33,10 +33,11 @@
 #include "PedData.h"
 #include "VoronoiDiagram.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/methods/Method_D.h:36]: Please include in cpp file.

File: jpsreport/methods/Method_I.cpp [hunk 21]
```diff
@@ -150,9 +151,9 @@ bool Method_I::Process(
         } else {
             if(IsPointsOnOneLine(XInFrame, YInFrame)) {
                 if(fabs(XInFrame[1] - XInFrame[0]) < dmin) {
-                    XInFrame[1] += offset;
+                    XInFrame[1] += joffset;
```
Comment (zeroset) [inline jpsreport/methods/Method_I.cpp:154]: Why is it joffset in method I? What is the J standing for?

File: jpsreport/methods/Method_I.h [hunk 22]
```diff
@@ -32,10 +32,11 @@
 #include "PedData.h"
 #include "VoronoiDiagram.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/methods/Method_I.h:35]: Please include in cpp file.

File: jpsreport/methods/Method_J.h [hunk 23]
```diff
@@ -32,10 +32,11 @@
 #include "PedData.h"
 #include "VoronoiDiagram.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/methods/Method_J.h:35]: Please include in cpp file.

File: jpsreport/methods/PedData.h [hunk 24]
```diff
@@ -29,14 +29,15 @@
 #ifndef PEDDATA_H_
 #define PEDDATA_H_
 
-#include "../IO/OutputHandler.h"
 #include "../general/Macros.h"
 #include "../tinyxml/tinyxml.h"
 
+#include <Logger.h>
```
Comment (zeroset) [inline jpsreport/methods/PedData.h:35]: please include in cpp file.

File: jpsreport/methods/PedData.h [hunk 25]
```diff
@@ -29,14 +29,15 @@
...
 #include <algorithm> // std::min_element, std::max_element
 #include <boost/algorithm/string.hpp>
 #include <boost/numeric/ublas/io.hpp>
 #include <boost/numeric/ublas/matrix.hpp>
+#include <fstream>
```
Comment (zeroset) [inline jpsreport/methods/PedData.h:40]: please include in cpp file.

File: jpsreport/methods/Method_I.cpp (hunk 21 above)
Comment (chraibi) [reply jpsreport/methods/Method_I.cpp:154]: **J**uPedSim

File: jpsreport/methods/Method_I.h (hunk 22 above)
Comment (JetteSchumann) [reply jpsreport/methods/Method_I.h:35]: Done

File: jpsreport/methods/Method_J.h (hunk 23 above)
Comment (JetteSchumann) [reply jpsreport/methods/Method_J.h:35]: Done

File: jpsreport/methods/Method_I.cpp (hunk 21 above)
Comment (JetteSchumann) [reply jpsreport/methods/Method_I.cpp:154]: renamed macro to JPS_OFFSET and moved it to Macros.h

File: jpsreport/methods/Method_A.h (hunk 17 above)
Comment (JetteSchumann) [reply jpsreport/methods/Method_A.h:36]: Done

File: jpsreport/methods/Method_B.h (hunk 18 above)
Comment (JetteSchumann) [reply jpsreport/methods/Method_B.h:35]: Done

File: jpsreport/methods/Method_C.h (hunk 19 above)
Comment (JetteSchumann) [reply jpsreport/methods/Method_C.h:34]: Done

File: jpsreport/methods/Method_D.h (hunk 20 above)
Comment (JetteSchumann) [reply jpsreport/methods/Method_D.h:36]: Done

File: jpsreport/methods/PedData.h (hunk 24 above)
Comment (JetteSchumann) [reply jpsreport/methods/PedData.h:35]: Done

File: jpsreport/methods/PedData.h (hunk 25 above)
Comment (JetteSchumann) [reply jpsreport/methods/PedData.h:40]: Done

File: jpsreport/geometry/Transition.h (hunk 15 above)
Comment (JetteSchumann) [reply jpsreport/geometry/Transition.h:34]: Done

File: jpsreport/geometry/Wall.h (hunk 16 above)
Comment (JetteSchumann) [reply jpsreport/geometry/Wall.h:34]: Done

File: jpsreport/geometry/Room.h (hunk 13 above)
Comment (JetteSchumann) [reply jpsreport/geometry/Room.h:33]: Done

File: jpsreport/geometry/SubRoom.h (hunk 14 above)
Comment (JetteSchumann) [reply jpsreport/geometry/SubRoom.h:35]: Done

File: jpsreport/geometry/Line.h (hunk 11 above)
Comment (JetteSchumann) [reply jpsreport/geometry/Line.h:34]: Done

File: jpsreport/geometry/Obstacle.h (hunk 12 above)
Comment (JetteSchumann) [reply jpsreport/geometry/Obstacle.h:32]: Done

File: jpsreport/geometry/Hline.h (hunk 10 above)
Comment (JetteSchumann) [reply jpsreport/geometry/Hline.h:35]: Done

File: jpsreport/geometry/Goal.h (hunk 9 above)
Comment (JetteSchumann) [reply jpsreport/geometry/Goal.h:34]: Done

File: jpsreport/geometry/Building.h (hunk 7 above)
Comment (JetteSchumann) [reply jpsreport/geometry/Building.h:39]: Done

File: jpsreport/geometry/Crossing.h (hunk 8 above)
Comment (JetteSchumann) [reply jpsreport/geometry/Crossing.h:35]: Done

File: jpsreport/general/ArgumentParser.h (hunk 6 above)
Comment (JetteSchumann) [reply jpsreport/general/ArgumentParser.h:36]: Done

File: jpsreport/CMakeLists.txt (hunk 5 above)
Comment (JetteSchumann) [reply jpsreport/CMakeLists.txt:41]: Done

File: jpsreport/Analysis.h (hunk 4 above)
Comment (JetteSchumann) [reply jpsreport/Analysis.h:41]: Done

**Pull request merged**